include/clang/Basic/TokenKinds.def

```diff
@@ -563,6 +563,11 @@ KEYWORD(__bridge_transfer            , KEYARC)
 KEYWORD(__bridge_retained            , KEYARC)
 KEYWORD(__bridge_retain              , KEYARC)
 
+
+// CHERI casts to and from capabilities
+KEYWORD(__cheri_cast                 , KEYALL)
```

Comment:
It would be nice to have an `__offset_cast` and `__address_cast` for capability-to-integer casts as well.

Should this be a keyword for all language dialects, or only for CHERI?  Ideally, it would be a normal identifier in other language dialects, though that's not essential because it's reserved for the implementation.